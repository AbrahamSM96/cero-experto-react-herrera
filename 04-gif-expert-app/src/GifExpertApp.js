import React, { useState } from 'react'
import PropTypes from 'prop-types'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball'])

    // const handleAdd = () => {
    //   setCategories(() => [...categories, 'Hulk'])
    // setCategories((cats) => [...cats, 'Hulk'])
    //  }

    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />
            <ol>
                {
                    categories.map((cat) =>
                        <GifGrid key={cat} category={cat} />
                    )
                }
            </ol>
        </div>
    )
}

GifExpertApp.protoTypes = {
    setCategories: PropTypes.func.isRequired
}