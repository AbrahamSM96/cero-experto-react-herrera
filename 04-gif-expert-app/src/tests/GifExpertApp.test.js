import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('test in componente GifExpertApp', () => {
    let wrapper = shallow(<GifExpertApp />)

    test('should show the componente correctly', () => {
        expect(wrapper).toMatchSnapshot()
    });
});