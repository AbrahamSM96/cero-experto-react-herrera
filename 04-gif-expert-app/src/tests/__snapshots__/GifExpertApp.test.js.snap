// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in componente GifExpertApp should show the componente correctly 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;
